### ISSUE TITLE
Compatibility with Noto build tools

### ISSUE THREAD

**Issue by C1710** [C1710/blobmoji#81]
All the new (and in an yet to be published branch almost all) emojis are not named `emoji_u<...>.svg` anymore, but are named well... by their official name.  
While this in my opinion really makes it easier to search for emojis and organize them, it does not work directly anymore with the scripts provided by Noto Emoji.

I already started working on a system for [`emoji_builder`](https://github.com/C1710/emoji_builder) (name still WIP) to render and rename the emojis such that the original scripts can be used instead of the (almost) identical ones built into the software itself, however right now this is low priority for me, so unless anyone needs this functionality, I will probably not implement it for now.

**Comment by DavidBerdik** [comment]
I would really like if there was a way to use the original Noto emoji scripts. Google release their final set for Unicode 14 today, and I would really like to compile a version of the font that includes the emojis from their set that are missing from your set until you finalize your set, but with this change, I suppose I will not be able to do that.

**Comment by C1710** [comment]
> I would really like if there was a way to use the original Noto emoji scripts. Google release their final set for Unicode 14 today, and I would really like to compile a version of the font that includes the emojis from their set that are missing from your set until you finalize your set, but with this change, I suppose I will not be able to do that.

I'm actually not quite sure what you mean with missing emojis? Emoji 14.0 is already fully supported for months now...

However, all assets in a Noto-compatible format are available for the current release 😅

**Comment by DavidBerdik** [comment]
> I'm actually not quite sure what you mean with missing emojis? Emoji 14.0 is already fully supported for months now...

Ah! My bad. I knew you had an Emoji 14.0 release, but I thought it was incomplete, so I was going to do a cross-compile like I did when people were waiting for your 13.1 release. After checking https://emojipedia.org/unicode-14.0/ on my phone though, I see that I was wrong. Sorry about that.

> However, all assets in a Noto-compatible format are available for the current release 😅

Well, since your Emoji 14.0 release is complete after all, I suppose that makes this irrelevant, but for the sake of future understanding, I thought you described that your current release is no longer compatible with the Noto scripts?

**Comment by C1710** [comment]
> Well, since your Emoji 14.0 release is complete after all, I suppose that makes this irrelevant, but for the sake of future understanding, I thought you described that your current release is no longer compatible with the Noto scripts?

The state of the repository is not directly compatible, that's correct. But for the [releases](https://github.com/C1710/blobmoji/releases/tag/v14.0.1), I build a compatible version (i.e. with the classic naming convention and prerendered PNGs) and add it to the release-page.  
My plan (although I don't have much time for this at the moment) is to add such a "conversion" feature to emoji_builder (well, I already have, but it's messy and not quite ready for publishing).

**Comment by DavidBerdik** [comment]
> The state of the repository is not directly compatible, that's correct. But for the [releases](https://github.com/C1710/blobmoji/releases/tag/v14.0.1), I build a compatible version (i.e. with the classic naming convention and prerendered PNGs) and add it to the release-page.

How are you currently handling conversions between builder and the Noto builder right now? Are you just doing it manually?

> My plan (although I don't have much time for this at the moment) is to add such a "conversion" feature to emoji_builder (well, I already have, but it's messy and not quite ready for publishing).

Sweet! Hopefully it's ready in time for Emoji 15! I haven't tried actually using it, but from briefly looking at it, I think it looks like a very promising tool.

**Comment by C1710** [comment]
> How are you currently handling conversions between builder and the Noto builder right now? Are you just doing it manually?

No, I actually just use emoji_builder, but with a different building pipeline - it doesn't build the actual fonts, but just renders the emojis (also with a different aspect ratio) and saves the renamed SVG files (which is, after all, the only difference).

**Comment by DavidBerdik** [comment]
> No, I actually just use emoji_builder, but with a different building pipeline - it doesn't build the actual fonts, but just renders the emojis (also with a different aspect ratio) and saves the renamed SVG files (which is, after all, the only difference).

Ah so it converts from your format to the Noto format, and then the Noto tools can be used to build the set?

> So that's what I meant with "I already have". It does work, I use it, but it's not very well done.

What are your concerns with how you currently do it? I haven't looked at your code thoroughly enough to really understand it that well or identify any flaws.

**Comment by C1710** [comment]
> Ah so it converts from your format to the Noto format, and then the Noto tools can be used to build the set?

Yeah, that's basically it. If I use the software directly to create the font, it does it similarly (in the end, the same build scripts are used), but skips some steps (like rendering to 128px, saving, compressing, saving again, etc.) and reuses some stuff that it has already done.

> What are your concerns with how you currently do it? I haven't looked at your code thoroughly enough to really understand it that well or identify any flaws.

Right now, that builder (which I haven't published) is just copy&pasted from the "normal" `Blobmoji` builder, but with some changes. And since I want to only publish clean code and not such a (working, but ugly and probably quite buggy) mess.  
Plus, I already plan some rather large changes which would require it to be rewritten in large parts, so I don't want to make everything clean and pretty just to have to do it all over again.

**Comment by DavidBerdik** [comment]
> Right now, that builder (which I haven't published, as far as I remember)

Ah, alright. I actually thought it had been published here: https://github.com/C1710/emoji_builder

**Comment by vkryl** [comment]
Some [Telegram X](https://github.com/TGX-Android/Telegram-X) user previously (and fan of blobmoji at the same time) requested to provide an update for the blobmoji emoji set, which currently fall backs to regular Noto for any emojis added after d72aafdcfcdf63c85a24b1eb126ad0c04b50942b commit, as it seems to be the last one to include [pre-rendered PNGs](https://github.com/C1710/blobmoji/tree/d72aafdcfcdf63c85a24b1eb126ad0c04b50942b/png/128). Is [emoji_builder](https://github.com/C1710/emoji_builder) currently the intended way to build the same archive for the apps that relied on it? Couldn't find any mentions of it in `README`. Telegram X is about to fully support Emoji 15.0, and could try to provide the way to use newer blobs for those who selected them in app settings.

**Comment by C1710** [comment]
> Some [Telegram X](https://github.com/TGX-Android/Telegram-X) user previously (and fan of blobmoji at the same time) requested to provide an update for the blobmoji emoji set, which currently fall backs to regular Noto for any emojis added after [d72aafd](https://github.com/C1710/blobmoji/commit/d72aafdcfcdf63c85a24b1eb126ad0c04b50942b) commit, as it seems to be the last one to include [pre-rendered PNGs](https://github.com/C1710/blobmoji/tree/d72aafdcfcdf63c85a24b1eb126ad0c04b50942b/png/128). Is [emoji_builder](https://github.com/C1710/emoji_builder) currently the intended way to build the same archive for the apps that relied on it? Couldn't find any mentions of it in `README`. Telegram X is about to fully support Emoji 15.0, and could try to provide the way to use newer blobs for those who selected them in app settings.

Sorry for the late answer; I have updated the most recent release to include a ZIP file with the rendered PNGs and I will also add one for upcoming releases: https://github.com/C1710/blobmoji/releases/tag/v15.0
It's the version that is rendered during the build of the font, so it includes flags and everything and might therefore be a bit messy.

Additionally t he latest version of the `main`-branch includes a small Python script that can rename the files to the `emoji_u<...>` format, which should make it much easier to use it together with Noto-based tools. I'm even considering to revert the new naming scheme altogether (except for the version that's currently under development, as it is easier for me to recognize the emojis by there names instead of their codepoints)
